packages/app/src/cli/services/dev/extension/payload.ts

```diff
@@ -24,6 +25,43 @@
     const {localization, status: localizationStatus} = await getLocalization(extension, options)
 
     const renderer = await getUIExtensionRendererVersion(extension)
+
+    const extensionPointsWithAssets = await Promise.all(
+      extension.configuration.extension_points.map(
+        async (extensionPoint: ExtensionInstance['configuration']['extension_points']) => {
+          const buildManifest = extensionPoint.build_manifest as {
+            assets: {[key: string]: {filepath: string}}
+          }
+
+          let assets: Asset = {}
+
+          if (buildManifest?.assets) {
+            const manifests = await Promise.all(
+              Object.entries(buildManifest.assets).map(async ([name, asset]) => ({
+                name,
+                url: `${url}${joinPath('/assets/', asset.filepath)}`,
+                lastUpdated:
+                  (await fileLastUpdatedTimestamp(joinPath(dirname(extension.outputPath), asset.filepath))) ?? 0,
+              })),
+            )
+            assets = manifests.reduce((acc, asset) => {
+              acc[asset.name] = asset
+              return acc
+            }, assets)
+
+            return {
+              ...extensionPoint,
+              assets,
+            }
+          }
+
+          return extensionPoint
+        },
+      ),
+    )
+
+    const extensionPoints = getExtensionPoints(extensionPointsWithAssets, url)
```

Comment:
A few proposals for this function:

- Use `UIExtensionSchemaType['extension_points']` to get the right types for the extension points
- By using the new Asset interface and a for loop, you can avoid the extra `reduce`

So the implementation is simplified quite a bit:

```ts
const extensionPointsTyped: UIExtensionSchemaType['extension_points'] = extension.configuration.extension_points
    const extensionPointsWithAssets = await Promise.all(
      extensionPointsTyped.map(async (extensionPoint) => {
        const buildManifest = extensionPoint.build_manifest
        if (!buildManifest.assets) return extensionPoint
        const assets: {[key: string]: Asset} = {}

        for (const [name, asset] of Object.entries(buildManifest.assets)) {
          assets[name] = {
            name,
            url: `${url}${joinPath('/assets/', asset.filepath)}`,
            // eslint-disable-next-line no-await-in-loop
            lastUpdated: (await fileLastUpdatedTimestamp(joinPath(dirname(extension.outputPath), asset.filepath))) ?? 0,
          }
        }

        return {...extensionPoint, assets}
      }),
    )
```

Reply:
One more thing, you might want to extract this into a proper function and call it inside `getExtensionPoints`, because `getExtensionPoints` is doing some validations on the type of the extension points, makes sense to apply this transform AFTER those validations.